test/realm_test.dart

```diff
@@ -1394,6 +1395,58 @@ Future<void> main([List<String>? args]) async {
     expect(await realmIsCancelled, isTrue);
     expect(progressReturned, isFalse);
   });
+
+  test('Realm - local realm can be compacted', () {
+    var config = Configuration.local([Car.schema]);
+    var realm = getRealm(config);
+    final compacted = realm.compact();
+    expect(compacted, true);
+  });
+
+  test('Realm - local realm can be compacted in worker isolate', () async {
```

Comment:
Should we provide a public API for this? Or perhaps this could be the only API ie 
Future<bool> compact()  // works in a worker isolate. This could trigger migration callbacks in that isolate, which could bring problems.

Reply:
I suggest we keep it as a test for now. It gets tricky quickly. For one realms cannot be passed between isolates since they hold references to native types.